driver-core/src/main/com/mongodb/internal/operation/MapReduceToCollectionOperation.java

```diff
@@ -595,8 +594,10 @@ public MapReduceStatistics apply(final BsonDocument result, final AsyncConnectio
 
     private BsonDocument getCommand(final ConnectionDescription description) {
         BsonDocument outputDocument = new BsonDocument(getAction(), new BsonString(getCollectionName()));
-        putIfTrue(outputDocument, "sharded", isSharded());
-        putIfTrue(outputDocument, "nonAtomic", isNonAtomic());
+        if (description != null && !serverIsAtLeastVersionFourDotFour(description)) {
```

Comment:
As these are ignored or can error in 4.4 I opted to exclude - especially as some error only in certain scenarios.

Reply:
Ack